File: Rotate Image.py
# https://leetcode.com/problems/rotate-image/
# Rotate Image

class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        new_matrix = []

        for i in range(len(matrix)):
            new_list = []
            for j in range(len(matrix)):
                j = len(matrix) - j -1
                new_list.append(matrix[j][i])

            new_matrix.append(new_list)

        for i in range(len(matrix)):
            matrix[i] = new_matrix[i]

File: Rotate Image.js
// https://leetcode.com/problems/rotate-image/?envType=daily-question&envId=2026-05-04
// Rotate Image

/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var rotate = function(matrix) {

    const n = matrix.length;
    const newMatrix = [];

    for (let i = 0; i < n; i++) {
        const newList = [];
        for (let j = 0; j < n; j++) {
            const newJ = n - j - 1;
            newList.push(matrix[newJ][i]);
        }
        newMatrix.push(newList);
    }

    for (let i = 0; i < n; i++) {
        matrix[i] = newMatrix[i];
    }
};